lib/request.js

```diff
@@ -15,7 +15,7 @@ const Transmit = require('./transmit');
 
 const internals = {
     events: Podium.validate(['finish', { name: 'peek', spread: true }, 'disconnect']),
-    reserved: ['server', 'url', 'query', 'path', 'method', 'mime', 'setUrl', 'setMethod', 'headers', 'id', 'app', 'plugins', 'route', 'auth', 'pre', 'preResponses', 'info', 'orig', 'params', 'paramsArray', 'payload', 'state', 'jsonp', 'response', 'raw', 'domain', 'log', 'logs', 'generateResponse']
+    reserved: ['server', 'url', 'query', 'path', 'method', 'mime', 'setUrl', 'setMethod', 'headers', 'id', 'app', 'plugins', 'route', 'auth', 'pre', 'preResponses', 'info', 'isInjected', 'orig', 'params', 'paramsArray', 'payload', 'state', 'jsonp', 'response', 'raw', 'domain', 'log', 'logs', 'generateResponse']
```

Comment:
This could _arguably_ be seen as a breaking change.  Maybe this should go into v20 for good measure CC @cjihrig.

Reply:
I don't think we should consider this semver major since the request object is owned by hapi. Any user code that is attaching things to the request object can use a symbol, `request.app`, etc.

Reply:
I was thinking of `server.decorate('request', ...)` usage, but either way I see your point.

Reply:
What you're saying makes sense. One thing we could do: automatically consider any additions to the `reserved` lists across the codebase as semver major.

Another longer term, but not mutually exclusive, idea: have `decorate()` update some new object (`request.decorations`?) on requests so that they won't ever conflict with hapi.